mlir-sycl/lib/Dialect/SYCL/IR/SYCLOps.cpp

```diff
@@ -259,6 +259,46 @@ LogicalResult SYCLIDConstructorOp::verify() {
       "expects a different signature. Check documentation for details");
 }
 
+static Type getBodyType(Type type) {
+  auto *ctx = type.getContext();
+  return TypeSwitch<Type, Type>(type)
+      .Case<HalfType>([&](auto) { return Float16Type::get(ctx); })
+      .Default({});
```

Comment:
optional: should we remove the default case, so we get assertion when there is an unexpected input type to `getBodyType`?

Reply:
The `areCastCompatible` methods on the (un)wrap ops use this helper to verify the op, so I think asserting here would be premature and would give us a less clear error message.